src/scripts/main.js

```diff
@@ -18,10 +18,34 @@ const food = {
   },
 };
 
+const treeDiv = document.createElement('div');
+
+treeDiv.id = 'tree';
+document.body.appendChild(treeDiv);
+
 const tree = document.querySelector('#tree');
 
 function createTree(element, data) {
-  // WRITE YOUR CODE HERE
+  if (element === null || data === undefined) {
+    throw new Error('arguments must not be null or undefined');
+  }
+
+  const ul = document.createElement('ul');
+
+  for (const key in data) {
+    const li = document.createElement('li');
+
+    li.textContent = key;
+
+    if (Object.entries(data[key]).length > 0) {
+      createTree(li, data[key]);
+    }
+    ul.appendChild(li);
+  }
+
+  if (ul.childNodes.length > 0) {
+    element.appendChild(ul);
+  }
 }
```

Comment:
This function is a bit complex and it's doing multiple things. It might be better to split it into a few smaller functions. For example, you could have one function for creating a list item, another for creating a list, and another for appending items to the list. This would make the code more modular and easier to understand.